### ISSUE TITLE
Improve angular velocity calculation

### ISSUE THREAD

**Issue by nimazareian** [UBC-Thunderbots/Software#2909]
### Description of the task
The new proportional controller used to calculate the angular velocity (added in #2896, implemented in `HRVOAgent::computeNewAngularVelocity`) seems to be working well during field testing. The main benefit of this new approach is that it is easier to configure a single value to make rotation work in real life (just by change the `ANGULAR_VELOCITY_KP`). 

However, this approach does not work as well as the previous implementation (which used 1D kinematic equations) in simulation. The robots rotate slower, specially when they only have to turn a few degrees, causing them to not reach their desired orientation fast enough. This is a problem for when the robot has to rotate to get possession of the ball, since the rotation will happen slower than the linear movement, causing the ball to hit the side of the robot, rather than the dribbler. 

The slower rotation is primarily due to the simlicity of the new implementation, which basically just takes the remaining amount which the robot has to rotate and multiplies it by a pre-defined constant, and uses the result as the new velocity. This approach can be problematic which the amount the robot has to rotate is small, resulting in the robot to rotate very slowly for the final few degrees. A similar problem was also faced for the linear velocity proportional controller added to HRVO, however, that issue was mitigated by introducing a dynamic proportional constant (kp) (the process of calculating this is documented on Notion). A similar approach could be used for the angular velocity as well, though it would be nice to have a more concrete solution that doesn't require tuning a bunch of values.
 

### Acceptance criteria

- [ ] Robots rotate fast in sim and receive the ball quickly without bumping into it
- [ ] Works well with robots in real life
- [ ] All tests disabled as a result of this issue should be re-enabled
- [ ] See if tests disabled for #2570 pass now

### Blocked By
The merge of #2896

**Comment by nimazareian** [comment]
With some further field testing, it turned out that the angular velocity kp can be increased further (2.8 -> 5.0), so the robots turn faster in sim as well, now.

**Comment by nimazareian** [comment]
This doesn't seem to be as big of an issue anymore. Closing for now. New planner should also improve this.
